python/pyspark/sql/readwriter.py

```diff
@@ -431,6 +432,8 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
         if column is not None:
             if numPartitions is None:
                 numPartitions = self._spark._sc.defaultParallelism
```

Comment:
I have a little worry whether this change will break the API. If some users just specify the `column`, `lowerBound`, `upperBound` in some Spark version, its program will fail after update, even very few people just use the default  parallelism. 

In my personal opinion, I prefer to make a change and  keep API consistent.

If your opinion is to add the assert on `numPartitions`, I will update the PR soon.